sky/packages/sky/lib/rendering/box.dart

```diff
@@ -450,6 +450,42 @@ abstract class RenderBox extends RenderObject {
     _size = inDebugBuild ? new _DebugSize(value, this, debugCanParentUseSize) : value;
   }
 
+  void applyPaintTransform(Matrix4 transform) {
+    if (parentData is BoxParentData) {
+      Point position = (parentData as BoxParentData).position;
+      transform.translate(position.x, position.y);
+    }
+  }
+
+  Point _transformPoint(Matrix4 transform, Point point) {
+    Vector3 position3 = new Vector3(point.x, point.y, 0.0);
+    Vector3 transformed3 = transform.transform3(position3);
+    return new Point(transformed3.x, transformed3.y);
+  }
```

Comment:
We should make this a free function because it doesn't use any state from |this|.